bot.py

```diff
@@ -0,0 +1,44 @@
+from telegram import Update
+from telegram.ext import ApplicationBuilder, CommandHandler, ContextTypes
+from exceptions import GameAlreadyExistError
+
+import interactions
+from config import token
+from tlgrm.utils import send_reply_message
+from orchestrator import Orchestrator
+
+
+# start the orchestrator at bot load
+orchestrator = Orchestrator()
+
+async def hello(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    if (update.effective_user):
+      await send_reply_message(update, f'Hello {update.effective_user.first_name}')
```

Comment:
Was thinking this code should go on ./app/bot.py ? or this is about something
else ?

Suggesting another update, let me know what you think.
Making sure the user is not a "bot" could be interesting
also for those check, across the codebase, i would suggess simple
decodaror to check for those valid info accross the user object, like:
```suggestion
from functools import wraps

# this could be as a list of decorator or just a generic class 
# class AccessControlRules that has a list of decorator functions, what do you
# think... ?
def check_user_is_not_bot(func):
    @wraps(func)
    async def wrapper(update, context): # abec avec les types... flemme :)
        if update.effective_user is not None and not update.effective_user.is_bot:
            return await func(update, context)
    return wrapper

@check_user_is_not_bot
async def hello(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    if first_name := getattr(update.effective_user, "first_name", None):
        await send_reply_message(
            update,
            f'Hello {first_name}',
        )
```

Reply:
With this way of setting decorator, we can set our own kind of rules for any telegram APIs... i don't know if the idea is good for you

Reply:
Interesting idea. Decorators will come in super handy to check if users are admins and superadmins etc.
We will have quite a good number of checks, plus ponyORM [creates db sessions from decorators](https://docs.ponyorm.org/transactions.html). I don't know if they can be easily nested. If yes, its a solid GO for me.

Only thing i'll change is the `is_not_bot` check.
[update.effective_user has an `is_bot` boolean](https://docs.python-telegram-bot.org/en/v20.5/telegram.user.html#telegram.User.params.is_bot) let's use it.

Reply:
About the location of the file,
Its more a matter of taste. I usually like main entrypoints on the root folder, it helps for discoverability.
I have no strong opinion about this. If you really feel like we should nest it. I'm fine with it too

Reply:
> About the location of the file,
> Its more a matter of taste. I usually like main entrypoints on the root folder, it helps for discoverability.

When deploying the project, we need to deploy without tests, and other dx stuffs on the root directory, a good way to do that is to have the source code logic inside a single dir "src", "app", "lamap-bot", idk... + can you tell me more about that "discoverability" please ?


> Only thing i'll change is the is_not_bot check.
> [update.effective_user has an is_bot boolean](https://docs.python-telegram-bot.org/en/v20.5/telegram.user.html#telegram.User.params.is_bot) let's use it.

For the "bot" check..., am not seeing the "is_not_bot" check you're talking about ? 🤔 or maybe a typo ? i added the "is_bot" among the decorator function already... let me know if you're talking about something else ?

The class I was talking about can be found here : https://github.com/dylantientcheu/lamap-bot/pull/106/commits/cdd41d8
(it's just an example btw)